### PR TITLE
Added nuspec and symbols.nuspec files to Razor.Design

### LINKED ISSUE #2215: Missing pdb inside Microsoft.AspNetCore.Razor.Design.symbols.nupkg
For some reason, the symbols nupkg we create for this assembly doesn't contain any pdb files.

cc @pranavkm

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Fixes #2215 

React to aspnet/BuildTools#630

We want to include the pdbs of the tasks and tools as part of the Razor.Design symbols package.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.symbols.nuspec [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.symbols.nuspec]: @pranavkm for the first pass (just to fix https://github.com/aspnet/Razor/issues/2215) can we just nuke this and include symbols in the regular NuspecFile? I noticed `dotnet pack` does the right thing. It automatically filters out pdbs from the main nupkg even if symbols are included in the nuspec file.
We can later add the symbols.nuspec file after we are ready to merge aspnet/BuildTools#630.

**Comment by ajaybhargavb** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.nuspec [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    <id>$id$</id>
+    <version>$version$</version>
+    <authors>$authors$</authors>
+    <description>$description$</description>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <tags>$tags$</tags>
+    <licenseUrl>$licenseUrl$</licenseUrl>
+    <projectUrl>$projectUrl$</projectUrl>
+    <iconUrl>$iconUrl$</iconUrl>
+    <repository type="git" url="$repositoryUrl$" />
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.nuspec]: Add commit as well. Example https://github.com/aspnet/IISIntegration/blob/7a3c260394f8689fccf15149454bd33170664248/src/Microsoft.AspNetCore.Server.IIS/Microsoft.AspNetCore.Server.IIS.nuspec#L16

File: src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj [hunk 3]
```diff
@@ -89,16 +86,41 @@
...
+        authors=$(Authors);
+        description=$(Description);
+        tags=$([MSBuild]::Escape($(PackageTags)));
+        licenseUrl=$(PackageLicenseUrl);
+        projectUrl=$(PackageProjectUrl);
+        iconUrl=$(PackageIconUrl);
+        repositoryUrl=$(RepositoryUrl);
+        copyright=$(Copyright);
+        targetframework=$(TargetFramework);
+        NetStandardLibraryVersion=$(NETStandardLibrary20PackageVersion);
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj]: Nit: trim unused properties like targetframework and NetStandardLibraryVersion

File: src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj [hunk 4]
```diff
@@ -89,16 +86,41 @@
...
+    <PropertyGroup>
+      <!-- Make sure we create a symbols.nupkg. -->
+      <IncludeSymbols>true</IncludeSymbols>
+
+      <NuspecProperties>
+        id=$(MSBuildProjectName);
+        version=$(PackageVersion);
+        authors=$(Authors);
+        description=$(Description);
+        tags=$([MSBuild]::Escape($(PackageTags)));
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj]: Nit: tags in nuspec should be space separated, so `$(PackageTags.Replace(';', ' '))`

File: src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj [hunk 5]
```diff
@@ -89,16 +86,41 @@
...
   </Target>
+
+  <Target Name="PopulateNuspec" BeforeTargets="GenerateNuspec" DependsOnTargets="BuiltProjectOutputGroup;DebugSymbolsProjectOutputGroup;_BuildDependencyProjects">
+
+    <PropertyGroup>
+      <!-- Make sure we create a symbols.nupkg. -->
+      <IncludeSymbols>true</IncludeSymbols>
+
+      <NuspecProperties>
+        id=$(MSBuildProjectName);
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj]: Nit: `$(PackageId)` - this happens to match MSBuildProjectName, but is a different value

**Comment by ajaybhargavb** [comment]
🆙 📅

**Comment by ajaybhargavb** [comment]
@natemcmaster Ping

**Pull request merged**